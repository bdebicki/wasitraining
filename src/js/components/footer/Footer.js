import LAYOUT from '../../elementHandlers/layout';
import CookiesInfo from './CookiesInfo';
import MoreInfo from './MoreInfo';

export default class Footer {
	static renderFooterContainer() {
		const footerContainer = document.createElement('footer');

		footerContainer.id = LAYOUT.FOOTER_ID;

		return footerContainer;
	}

	static render() {
		const footerContainer = Footer.renderFooterContainer();

		footerContainer.appendChild(CookiesInfo.render());
		footerContainer.appendChild(MoreInfo.render());

		return footerContainer;
	}
}
