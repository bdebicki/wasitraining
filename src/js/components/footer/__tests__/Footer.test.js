import Footer from '../Footer';

describe('tests for footer component', () => {
  it('render footer component', () => {
    // when
    const footerEl = Footer.render();

    // then
    expect(footerEl).toMatchSnapshot();
  });
});
