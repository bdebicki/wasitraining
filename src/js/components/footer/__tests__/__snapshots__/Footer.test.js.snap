// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for footer component render footer component 1`] = `
<footer
  id="footer"
>
  <p
    class="cookiesSection"
  >
    yep. this site uses cookies. click 
    <a
      class="link"
      href="#cookies-policy"
    >
      here
    </a>
     for more.
  </p>
  <section
    class="aboutSection"
  >
    <p
      class="aboutSection__copyrights"
    >
      made with ‚ù§ in Gdynia 2017
    </p>
    <a
      class="link link--inverted aboutSection__aboutLink"
      href="#about"
    >
      about
    </a>
  </section>
</footer>
`;
